src/index.js

```diff
@@ -334,9 +334,8 @@ function ncc (
               transpileOnly,
               compiler: eval('__dirname + "/typescript.js"'),
               compilerOptions: {
+                ...fullTsconfig.compilerOptions,
```

Comment:
We should still put `module: 'esnext'` as the default, before this line so that it gets overridden:

```suggestion
                module: 'esnext'
                ...fullTsconfig.compilerOptions,
```